lib/vanagon/component/source/local.rb

```diff
@@ -148,7 +148,11 @@ def decompressor
         # @return [String] the directory that should be traversed into to build this source
         # @raise [RuntimeError] if the @extension for the @file isn't currently handled by the method
         def dirname
-          archive? ? File.basename(file, extension) : './'
+          if archive? || File.directory?(file)
+            File.basename(file, extension)
```

Comment:
`file` should always be a `Pathname` object; that means you can condense this a little bit:

```ruby
if archive? || file.directory?
  file.basename(extension)
```